scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/synchronization/Synchronization.java

```diff
@@ -0,0 +1,1130 @@
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.objectweb.proactive.annotation.PublicAPI;
+
+
+/**
+ * Synchronization provides a persistent <i>key/value store</i> API, similar to the {@link java.util.Map Map} interface.
+ *
+ * The Synchronization service is automatically started with the ProActive Server.
+ *
+ * <p>The key/value store is organized in <i>channels</i>. Each channel corresponds to a private key/value {@link java.util.HashMap HashMap}.
+ * Synchronization provides an API to create or delete channel and decide if the channel should be persisted on disk (thus available on scheduler restart) or kept in memory only.</p>
+ *
+ * <p>Each map operation is performed on a given <i>channel</i>, with the channel name provided as parameter.</p>
+ *
+ * <p>Each map operation can be performed <b>remotely</b>, for example from <i>ProActive Workflow Tasks</i> executed inside <i>remote ProActive Nodes</i>.</p>
+ *
+ * <p>Each operation is as well <b>synchronous</b> and <b>atomic</b>, remote operations are received by the service and handled through a request queue.</p>
+ *
+ * <p>Java 8 Map operations, based on lambdas, are allowed, but lambdas must be provided as String values containing <a href="http://groovy-lang.org/closures.html">Groovy Closures</a> expressions which match the corresponding java lambda.</p>
+ *
+ * <p>For example, to write a groovy closure which match a Java {@link java.util.function.BiFunction}:</p>
+ *
+ * <pre> {@code
+ * "{ k, x -> x + 1 }"
+ * }</pre>
+ *
+ * <p>The Synchronization API provides also additional map operations such as {@link #conditionalCompute conditionalCompute} to perform conditional operations based on the result of a {@link java.util.function.BiPredicate}, or wait operations such as {@link #waitUntil(String, String, String)} to block the caller until a given predicate is met.</p>
+ *
+ * <p>These <i>wait operations</i> are the core of the Synchronization API and are mostly used to synchronize parallel ProActive Tasks. </p>
+ *
+ * @author ActiveEon Team
+ * @since 27/03/2018
+ * @see java.util.Map
+ * @see <a href="http://groovy-lang.org/closures.html">Groovy Closures</a>
+ */
+@PublicAPI
+public interface Synchronization extends Serializable {
+
+    // Channel Operations
+
+    /**
+     * Create a channel, overwrite any already existing channel with the same name
+     *
+     * @param name new channel name
+     * @param isPersistent controls whether the channel should have a database backup (in case of crash or restart) or is kept in memory only
+     * @return true if the channel was created and no previous channel with the same name already existed
+     * @throws IOException if an error occurred while persisting the channel on disk
```

Comment:
Users of this API might think that they can expect no IOException if they put `isPersistent` to `false`. Is that correct?

Reply:
yes that's correct.